components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/DefaultSolrVocabularyResourceManager.java

```diff
@@ -59,16 +73,84 @@
     @Inject
     private CacheManager cacheFactory;
 
+    @Inject
+    private Environment environment;
+
     @Override
     public void initialize(String vocabularyName) throws InitializationException
     {
+        // Get jars home path to where the jar resources are stored
+        Class<DefaultSolrVocabularyResourceManager> clazz = DefaultSolrVocabularyResourceManager.class;
+        String jarPath = clazz.getProtectionDomain().getCodeSource().getLocation().getPath();
+        String jarsHome = jarPath.substring(0, jarPath.lastIndexOf('/'));
+        // Get data Solr home path
+        File solrHome = new File(this.environment.getPermanentDirectory().getAbsolutePath(), "solr");
+        File dest = solrHome;
+
+        CoreContainer container = this.cores.getContainer();
+
+        // Check if the core doesn't exist already
+        if (container.getCore(vocabularyName) != null) {
+            return;
+        }
+
         try {
-            this.core = new EmbeddedSolrServer(this.cores.getContainer(), vocabularyName);
+            File dir = new File(jarsHome);
+            File[] directoryListing = dir.listFiles();
+
+            if (directoryListing == null) {
+                return;
+            }
+
+            for (File file : directoryListing) {
+                JarFile jar = new JarFile(file);
+                String jarName = jar.getName().substring(jarsHome.length());
+                if ("jar".equals(FilenameUtils.getExtension(file.toPath().toString()))
+                    && jarName.startsWith("vocabulary-" + vocabularyName)) {
```

Comment:
Do we ever have multiple jars with the same vocabulary name, from which we want to copy resources? Maybe its better to be more selective, and extract the jar location from specific classes, so we don't accidentally extract some files from a jar that matched this pattern? For example, if we have two versions of vocabularies.